devel/githubContrib.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// go run githubContrib.go to update Contributions.txt
+
+// This script goes from org -> repos (skipping forks) -> contributors -> user
+// -> guess/normalize the company and count contribs
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Check checks for non nil error and dies upon error
+func Check(err error, msg string) {
+	if err != nil {
+		log.Fatal(msg, err)
+	}
+}
+
+// getToken gets auth token from the env
+func getToken() string {
+	// fmt.Println("in GetToken")
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		log.Fatal("Need to have GITHUB_TOKEN set in the env")
+	}
+	return token
+}
+
+var gAuthToken = getToken() // Any other way to get a static ?
```

Comment:
got rid of that cache, doesn't matter in this case given we do http requests and it hinders testability of the file